datafusion/substrait/src/logical_plan/producer.rs

```diff
@@ -1686,7 +1691,7 @@ fn to_substrait_bounds(window_frame: &WindowFrame) -> Result<(Bound, Bound)> {
     ))
 }
 
-fn to_substrait_literal(value: &ScalarValue) -> Result<Literal> {
+fn to_substrait_literal(value: &ScalarValue, nullable: bool) -> Result<Literal> {
```

Comment:
According to [ExprSchemable::nullable()](https://github.com/apache/datafusion/blob/3e4e09a7b086009921588d52be577ada0ed6d6b9/datafusion/expr/src/expr_schema.rs#L267), the nullability of a literal depends on its `ScalarValue`,  so I think the `nullable` argument for this function could be omitted.
```rust
fn to_substrait_literal(value: &ScalarValue, nullable: bool) -> Result<Literal> {
    if value.is_null() {
        return Ok(Literal {
            nullable: true,
            type_variation_reference: DEFAULT_TYPE_REF,
            literal_type: Some(LiteralType::Null(to_substrait_type(
                &value.data_type(),
                true,
            )?)),
        });
    }
```
At the same time, we can also remove `try_to_substrait_null`.

Reply:
I think you're right, thanks! That does indeed simplify things. Fixed in [cbd2264](https://github.com/apache/datafusion/pull/10640/commits/cbd22641d3673beec2cb045d658f8c585724b793)